test/shared/connectors/BaseDownstreamConnectorSpec.scala

```diff
@@ -273,102 +223,227 @@ class BaseDownstreamConnectorSpec extends ConnectorSpec {
+              .put(
+                absoluteUrl,
+                config = dummyHeaderCarrierConfig,
+                body,
+                requiredHeaders = requiredTysIfsHeaders ++ List("Content-Type" -> "application/json"),
+                excludedHeaders = List(userContentType)
+              )
+              .returns(Future.successful(outcome))
 
-      "get" must {
-        "get with the required des headers and return the result" in new TysIfsLocalTest {
-          implicit val hc: HeaderCarrier = HeaderCarrier(otherHeaders = otherHeaders ++ List("Content-Type" -> "application/json"))
-
-          MockedHttpClient
-            .get(
-              absoluteUrl,
-              config = dummyHeaderCarrierConfig,
-              parameters = qps,
-              requiredHeaders = requiredTysIfsHeaders,
-              excludedHeaders = List("AnotherHeader" -> "HeaderValue")
-            )
-            .returns(Future.successful(outcome))
-
-          await(connector.get(TaxYearSpecificIfsUri[Result](url), queryParams = qps)) shouldBe outcome
-        }
+            await(connector.put(body, TaxYearSpecificIfsUri[Result](url))) shouldBe outcome
+          }
       }
+    }
+  }
 
-      "delete" must {
-        "delete with the required des headers and return the result" in new TysIfsLocalTest {
-          implicit val hc: HeaderCarrier = HeaderCarrier(otherHeaders = otherHeaders ++ List("Content-Type" -> "application/json"))
+  "passThroughHeaders()" when {
+    import scala.language.reflectiveCalls
 
-          MockedHttpClient
-            .delete(
-              absoluteUrl,
-              config = dummyHeaderCarrierConfig,
-              requiredHeaders = requiredTysIfsHeaders,
-              excludedHeaders = List("AnotherHeader" -> "HeaderValue"))
-            .returns(Future.successful(outcome))
+    val connector = new BaseDownstreamConnector with MockAppConfig with MockHttpClient {
+      val http: HttpClient     = mockHttpClient
+      val appConfig: AppConfig = mockAppConfig
 
-          await(connector.delete(TaxYearSpecificIfsUri[Result](url))) shouldBe outcome
-        }
+      def checkPassThroughHeaders(downstreamConfig: DownstreamConfig, additionalHeaders: Seq[(String, String)]): Seq[(String, String)] =
+        passThroughHeaders(downstreamConfig, additionalHeaders)
+    }
+
+    "given some environmentHeaders and additionalHeaders" should {
+
+      "return the expected passthrough headers" in {
+        val result = connector.checkPassThroughHeaders(
+          DownstreamConfig(
+            "baseUrl",
+            "env",
+            "token",
+            environmentHeaders = Some(List("Gov-Test-Scenario", "AnotherHeader"))
+          ),
+          additionalHeaders = List(
+            "AnotherHeader" -> "expect-to-be-filtered",
+            "not-in-hc"     -> "expect-to-be-filtered"
+          )
+        )
+
+        result shouldBe List(
+          "Gov-Test-Scenario" -> "DEFAULT"
+        )
```

Comment:
For readability (and conciseness) would it be worth:

- Moving an implicit header config closer (it's not clear in this case where the DEFAULT GTS is coming from) e.g. by making the incoming request headers it contains an (explicit) parameter of `checkPassThroughHeaders`.
- Pulling out the boilerplate/common `DownstreamConfig` bits into the overridden `checkPassThroughHeaders`